sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -49,7 +49,7 @@ private[sql] class SessionState(sparkSession: SparkSession) {
   lazy val conf: SQLConf = new SQLConf
 
   def newHadoopConf(): Configuration = {
-    val hadoopConf = new Configuration(sparkSession.sharedState.hadoopConf)
+    val hadoopConf = new Configuration(sparkSession.sparkContext.hadoopConfiguration)
```

Comment:
@zsxwing When we call `newHadoopConf ` at here, we are about to launch a job to read/write data. So, it is fine.